e2e/pages/adf/contentServicesPage.ts

```diff
@@ -229,459 +221,396 @@ export class ContentServicesPage {
+    async sortByCreated(sortOrder: string): Promise<any> {
+        await this.contentList.dataTable.sortByColumn(sortOrder, this.createdColumnHeader);
     }
 
-    sortAndCheckListIsOrderedByName(sortOrder: string) {
-        this.sortByName(sortOrder);
-        const deferred = protractor.promise.defer();
-        this.checkListIsSortedByNameColumn(sortOrder).then((result) => {
-            deferred.fulfill(result);
-        });
-        return deferred.promise;
+    async sortAndCheckListIsOrderedByName(sortOrder: string): Promise<any> {
+        await this.sortByName(sortOrder);
+        return await this.checkListIsSortedByNameColumn(sortOrder);
     }
 
-    async checkListIsSortedByNameColumn(sortOrder: string) {
+    async checkListIsSortedByNameColumn(sortOrder: string): Promise<any> {
         return await this.contentList.dataTablePage().checkListIsSorted(sortOrder, this.columns.name);
     }
 
-    async checkListIsSortedByCreatedColumn(sortOrder: string) {
+    async checkListIsSortedByCreatedColumn(sortOrder: string): Promise<any> {
         return await this.contentList.dataTablePage().checkListIsSorted(sortOrder, this.columns.created);
     }
 
-    async checkListIsSortedByAuthorColumn(sortOrder: string) {
+    async checkListIsSortedByAuthorColumn(sortOrder: string): Promise<any> {
         return await this.contentList.dataTablePage().checkListIsSorted(sortOrder, this.columns.createdBy);
     }
 
-    async checkListIsSortedBySizeColumn(sortOrder: string) {
+    async checkListIsSortedBySizeColumn(sortOrder: string): Promise<any> {
         return await this.contentList.dataTablePage().checkListIsSorted(sortOrder, this.columns.size);
     }
 
-    sortAndCheckListIsOrderedByAuthor(sortOrder: string) {
-        this.sortByAuthor(sortOrder);
-        const deferred = protractor.promise.defer();
-        this.checkListIsSortedByAuthorColumn(sortOrder).then((result) => {
-            deferred.fulfill(result);
-        });
-        return deferred.promise;
+    async sortAndCheckListIsOrderedByAuthor(sortOrder: string): Promise<any> {
+        await this.sortByAuthor(sortOrder);
+        return await this.checkListIsSortedByAuthorColumn(sortOrder);
     }
 
-    sortAndCheckListIsOrderedByCreated(sortOrder: string) {
-        this.sortByCreated(sortOrder);
-        const deferred = protractor.promise.defer();
-        this.checkListIsSortedByCreatedColumn(sortOrder).then((result) => {
-            deferred.fulfill(result);
-        });
-        return deferred.promise;
+    async sortAndCheckListIsOrderedByCreated(sortOrder: string): Promise<any> {
+        await this.sortByCreated(sortOrder);
+        return await this.checkListIsSortedByCreatedColumn(sortOrder);
     }
 
-    doubleClickRow(nodeName) {
-        this.contentList.doubleClickRow(nodeName);
-        return this;
+    async doubleClickRow(nodeName): Promise<void> {
+        await BrowserActions.closeMenuAndDialogs();
```

Comment:
this should be the responsibility of the test
this method should have a single responsibility: double clicking on an element.